[Shezad Sulyman Imran Hassan] iP

SHEZADHASSAN22 opened the pull request:
# DukePro

*DukePro* frees your mind of having to remember things you need to do. It's,
* test based
* easy to learn
* ~~fast~~ Super Fast to use

All you need to do it,
1. download it from [here](http://github.com)
2. double click it.
3. add your tasks,
4. let it manage your tasks for you!

And it is **Free**

Here is the `main` method:
```java
public static void main(String[] args) {
        new Ui().greeting();
        new Duke("dukeData.txt").run();
    }
```

Features:
- [x] Managing tasks
- [ ] Managing Deadlines (coming soon) 💯 

As Kanye West said:
> We're living the future so
> the present is our past.

Commit:
Add Gradle support

SHEZADHASSAN22 committed:
Level 1 - Greet, Echo, Exit

SHEZADHASSAN22 committed:
Level 2 - Add, List

SHEZADHASSAN22 committed:
Level 3 - Mark as Done

SHEZADHASSAN22 committed:
Level 4 - Added ToDo, Event, Deadline

SHEZADHASSAN22 committed:
Level 5 - Automated Text UI Testing

SHEZADHASSAN22 committed:
Level 5 and 6. Handled errors and added delete function

SHEZADHASSAN22 committed:
Added more error handling for individual function

SHEZADHASSAN22 committed:
More error handling in add and delete functions

SHEZADHASSAN22 committed:
changed duke name in intro and cleaned up

SHEZADHASSAN22 committed:
Save data into hard disk

the users data was not being stored and so it would be lost on shutting down of the program.

The data is now stored on the hard disk and so the data is not lost on different sessions

SHEZADHASSAN22 committed:
Change date variables from Strings to java.tim.LocalDateTime class

Dates were being stored as Strings which is inconvenient in the long run. Fromat of the dates and times required in input is changed to be more readable when displayed

SHEZADHASSAN22 committed:
Merge Level 7  master branch to add ability to save to hard disk

Data was not being stored on hard disk and was lost on closing of the program.

SHEZADHASSAN22 committed:
Changed dates from Strings to LocalDateTime

Dates were represented by strings which may be inconvenient. Now they are of class LocalDateTime which are easier to work with.

SHEZADHASSAN22 committed:
Seperete logic into different classes for a more OOP approach

The logic was seperated into several classes that dealt with specific jobs.

SHEZADHASSAN22 committed:
Divide classes into packages

Created packages to make it more organised and reusable.

SHEZADHASSAN22 committed:
Add JUnit tests

SHEZADHASSAN22 committed:
Added JavaDoc Comments

SHEZADHASSAN22 committed:
Tweak code to comply with coding standards

SHEZADHASSAN22 committed:
Give users a way to find a task by searching for a keyword

SHEZADHASSAN22 committed:
Add JavaDocs to Code

SHEZADHASSAN22 committed:
Tweak the code to comply with a coding standard

SHEZADHASSAN22 committed:
Add ability to find tasks with user given search word

Finished merging 3 branches to add JavaDocs, clean up the code and add search functionality to allow user to search for a task

SHEZADHASSAN22 committed:
Package the app as a JAR File

SHEZADHASSAN22 committed:
Adding gradle support

darrenhoon commented:
Your header **DukePro** looks great!

```
new Ui().greeting();
```
hey, I like how clean this bit of code is 👍

Regarding "Kanye West", perhaps it should be "Kanye _East_"?

darrenhoon commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,22 @@
+
+import java.time.LocalDateTime;
+
+public class Event extends Task{
```
Perhaps there could be a header comment here to help users understand what this code does?

```
/**
 * The Event class is a child class of Task.
 * Contains additional "date" string parameter.
 * @author  
 * @version 
 * @since  
 */
```

Similar comment for the other classes to ensure you don't lose marks unnecessarily due to CheckStyle issues :)

SHEZADHASSAN22 replied:
Cool thanks bro.

darrenhoon commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+        }
+    }
+
+    public List<Task> getTasks() {
```
missing header comment here for method

darrenhoon commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -5,3 +5,39 @@ Hello from
 | |_| | |_| |   <  __/
 |____/ \__,_|_|\_\___|
 
+What can I do for you?
```
Great to see that you are doing tests! Your test-driven development greatly helps to reduce the number of bugs that you might run into in the future, which can save time. Great job!

darrenhoon commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * Prints the list of todos sequentially.
+     *
```
Perhaps the header comments should not have empty lines based on JavaDocs convention?

darrenhoon reviewed (commented):
Overall, the structure looks good as it helps to ensure modularity, such as Event, Todo, and Deadline inheriting from Task.

Testing shows great effort from you to ensure your code's rigidity against fringe cases while meaningful names ensure that anyone who wants to join your project can easily follow along to understanding the code and quickly improve the onboarding process.

SHEZADHASSAN22 committed:
Converted duke from CLI app to a GUI app

JavaFX used to implement a GUI. Previous UI class and methods removed as no longer used.

SHEZADHASSAN22 committed:
Merged branch-Level-10 to change from CLI to GUI app

Removed UI class as no longer needed.

SHEZADHASSAN22 committed:
Add assertions to several files

SHEZADHASSAN22 committed:
Improve code quality

Code included deeply nested if statements and long methods.

Code was seperated into several methods to organise the logic and improve overall quality

SHEZADHASSAN22 committed:
Merge pull request #1 from SHEZADHASSAN22/branch-A-Assertions

Add assertions to several files

SHEZADHASSAN22 committed:
Merge branch 'master' into branch-A-CodeQuality

SHEZADHASSAN22 committed:
Merge pull request #2 from SHEZADHASSAN22/branch-A-CodeQuality

Improve code quality

SHEZADHASSAN22 committed:
rearrange assert method

SHEZADHASSAN22 committed:
Add ability to delete and markAsDone multiple tasks in one command

SHEZADHASSAN22 committed:
Add ability to delete and markAsDone several tasks in one command

Could only delete and markAsDone a single task but now the user can insert more tasks to delete in the same command separated
by spaces to indicate several tasks to perform the action on.

SHEZADHASSAN22 committed:
improve the GUI

SHEZADHASSAN22 committed:
Use Checkstyle to clean up code

SHEZADHASSAN22 committed:
Add User Guide

SHEZADHASSAN22 committed:
Change Picture in docs

SHEZADHASSAN22 committed:
Change UI.png name to Ui.png

SHEZADHASSAN22 committed:
Create a new Jar file

CheahYan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,117 @@
+    /**
+     * Represents task in the from that it is stored in the data file.
+     *
+     * @return String of the task in the form required by teh data file.
```
I think there is a typo for your comment here. for "teh"

CheahYan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,44 @@
+     * @param userInput Input from user.
+     * @return False if user wants to close the program and true otherwise.
+     */
+    public String command(String userInput) {
+        String toReply;
+        if (userInput.equalsIgnoreCase("bye")) {
+            toReply = "Bye. Hope to see you again soon!";
+
+        } else if (userInput.equalsIgnoreCase("list")) {
+            toReply = taskList.showTasks(taskList.getTasks());
+        } else if (userInput.toLowerCase().startsWith("done")) {
+            toReply = taskList.markTaskDone(userInput);
+            storage.updateDataSet(taskList.getTasks());
+        } else if (userInput.toLowerCase().startsWith("delete")) {
+            toReply = taskList.deleteTask(userInput);
+            storage.updateDataSet(taskList.getTasks());
+        } else if (userInput.toLowerCase().startsWith("find")) {
+            toReply = taskList.findTask(taskList.getTasks(), userInput);
+        } else {
+            toReply = taskList.addTask(userInput, storage.getDataPath());
+        }
```
Do you think this may be too wordy for one method? Perhaps having some methods to abstract out some  code will make it more readable?

CheahYan reviewed (commented):
Overall, I find your code very easy to read and it is clean for the most part. It also follows the coding standards very strictly and I was not able to spot many mistakes. LGTM!

CheahYan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
```
Missing header comment here for your class!

CheahYan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,117 @@
+import java.time.LocalDateTime;
+
+
+public class Task {
```
Missing header comments here too for your class.

CheahYan reviewed (commented):
Updates to my previous peer review. Realised that I did not looked at Javadoc-related issues. There are missing header comments for some of your classes. Otherwise, the code follows the coding standards set by the module. Good job! LGTM!

SHEZADHASSAN22 committed:
Changed SDK
